### DISCUSSION TITLE
Cannot do OP logout for OKTA

### DISCUSSION THREAD

**Discussion by r-kotagudem** [OpenIDC/mod_auth_openidc#812] [Q&A] [answered]
Okta op logout required to have a query parameter id_token_hint as below including post_logout_redirect_uri
https://dev-<>.okta.com/oauth2/default/v1/logout?post_logout_redirect_uri=<redirect_uri>&id_token_hint=<id token>

However, triggering below redirect to OP where id_token_hint will be missing causing error

http://ip/oauth/callback?logout=<redirect_uri>&id_token_hint=<idtoken>

Could you please suggest a solution to do a complete logout (local logout and OP logout)?

**Comment by zandbelt** [comment, marked as answer]
Can you provide the server debug logs leading up to this?

**Reply by rahulkumarcisco** [reply]
Hi Zanbelt,

We are also facing similar issue while trying OP logout when OIDCSessionType is set to "client-cookie". Please find debug logs attached here. My mod_auth_openidc version is 2.4.11.
[debug_logs_okta_oidc.zip](https://github.com/zmartzone/mod_auth_openidc/files/9269915/debug_logs_okta_oidc.zip)
1

**Reply by zandbelt** [reply]
You have to use a server side caching type for the id token to be stored for this purpose

**Reply by rahulkumarcisco** [reply]
Do you mean there is a way to use server side caching with OIDCSessionType set to "client-cookie" instead of "server-cache"? Could you explain the config details to achieve that? We have requirement to use client_cookie mode to store mod_auth_oidc module session.

**Reply by zandbelt** [reply]
no, you need to use `server-cache`

**Reply by rahulkumarcisco** [reply]
We have multiple clustered apache instances running behind a LB, so using server-cache isn't possible as the OIDC session information isn't shared across multiple Apache instances. That's why we had to use 'client-cookie' for session storage but the logout fails with this setting. Could you suggest best solution for this scenario?

**Reply by rahulkumarcisco** [reply]
Also raised Issue https://github.com/zmartzone/mod_auth_openidc/issues/908

**Comment by zandbelt** [comment]
for the record: support for storing the id_token in a client-cookie based session was added in 2.4.12 https://github.com/zmartzone/mod_auth_openidc/commit/99373b612c4074610c46b22392070d2a5b508780
